eth_account/typed_transactions/set_code_transaction.py

```diff
@@ -0,0 +1,289 @@
+
+    @classmethod
+    def from_bytes(cls, encoded_transaction: HexBytes) -> "SetCodeTransaction":
+        """Builds an SetCodeTransaction from a signed encoded transaction."""
+        if not isinstance(encoded_transaction, HexBytes):
+            raise TypeError(f"expected Hexbytes, got type: {type(encoded_transaction)}")
+        if not (
+            len(encoded_transaction) > 0
+            and encoded_transaction[0] == cls.transaction_type
+        ):
+            raise ValueError("unexpected input")
+        # Format is (0x04 || TransactionPayload)
+        # We strip the prefix, and RLP unmarshal the payload into our
+        # signed transaction serializer.
+        transaction_payload = encoded_transaction[1:]
+        rlp_serializer = cls._signed_transaction_serializer
+        dictionary = rlp_serializer.from_bytes(transaction_payload).as_dict()  # type: ignore  # noqa: E501
+        rpc_structured_dict = transaction_rlp_to_rpc_structure(dictionary)
+        rpc_structured_dict["type"] = cls.transaction_type
+        return cls.from_dict(rpc_structured_dict)
+
+    def as_dict(self) -> Dict[str, Any]:
+        """Returns this transaction as a dictionary."""
+        dictionary = self.dictionary.copy()
+        dictionary["type"] = self.__class__.transaction_type
+        return dictionary
+
+    def hash(self) -> bytes:
+        """
+        Hashes this SetCodeTransaction to prepare it for signing.
+        As per the EIP-7702 specifications, the signature is a secp256k1 signature over
+        ``keccak256(0x04 || rlp([chainId, nonce, max_priority_fee_per_gas,
+        max_fee_per_gas, gasLimit, gasLimit, to, value, data, accessList,
+        authorizationList])).``
+        """
+        # Remove signature fields.
+        transaction_without_signature_fields = dissoc(self.dictionary, "v", "r", "s")
+        # RPC-structured transaction to rlp-structured transaction
+        rlp_structured_txn_without_sig_fields = transaction_rpc_to_rlp_structure(
+            transaction_without_signature_fields
+        )
+        rlp_serializer = self.__class__._unsigned_transaction_serializer
+        hash_ = pipe(
+            rlp_serializer.from_dict(rlp_structured_txn_without_sig_fields),  # type: ignore  # noqa: E501
+            lambda val: rlp.encode(val),  # rlp([...])
+            lambda val: bytes([self.__class__.transaction_type])
+            + val,  # (0x04 || rlp([...]))  # noqa: E501
+            keccak,  # keccak256(0x04 || rlp([...]))
+        )
+        return cast(bytes, hash_)
+
+    def payload(self) -> bytes:
+        """
+        Returns this transaction's payload as bytes.
+
+        Here, the transaction payload is:
+
+            TransactionPayload = rlp([chainId,
+            nonce, maxPriorityFeePerGas, maxFeePerGas, gasLimit,
+            to, value, data,
+            accessList,
+            authorizationsList,
+            signatureYParity, signatureR, signatureS])
```

Comment:
Did we decide this should be `y_parity, r, s` or am I thinking of something else?

Reply:
The payload definition in the EIP makes this distinction. I think this is fine. I'd just rather not have the actual fields named `signature...` or `signer...` and have them instead be `r`, `s`, `y_parity` / `yParity`, as both the EIP and our code currently do.